src/borg/testsuite/repository.py

```diff
@@ -15,11 +15,39 @@
 from ..remote import RemoteRepository, InvalidRPCMethod, PathNotAllowed
 from ..repository import Repository, LoggedIO, MAGIC, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
 from ..repoobj import RepoObj
-from . import BaseTestCase
 from .hashindex import H
 
 
-UNSPECIFIED = object()  # for default values where we can't use None
+class SingularRepositoryMaker:
+    # For now, this class is not used. Just a placeholder until I am sure if/how to incorporate it into project.
+    def __init__(self, exclusive=True, create=True):
+        self.current_repo = None
+        self.path = os.path.join(tempfile.mkdtemp(), "repository")
+        self.exclusive = exclusive
+        self.create = create
+        self.exit_stack = ExitStack()
+
+    def __enter__(self):
+        if self.current_repo is not None:
+            raise RuntimeError("Cannot re-enter a disposed repository.")
+        self.current_repo = self.exit_stack.enter_context(Repository(self.path, self.exclusive, self.create))
+        return self.current_repo
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        # Decide what needs to be done here.
+        # self.context_stack.undo()  # perhaps pass over args here
+        self.current_repo = None
```

Comment:
Per @RonnyPfannschmidt's suggestion, however I'm not sure how to incorporate this into what we have yet, but wanted to leave in for ideas/suggestions. Currently this class is not used.

Reply:
To pick up on @RonnyPfannschmidt's comment regarding this code: I think he's right in that the `__enter/__exit` stuff isn't very clear. But as Ted says, this code isn't used right now.

If that's the case, the Pytest `repository` fixture and the context manager seem to do the same job and we can get rid of one of them. In detail, the fixture an already run some code before and after a test and `__enter/__exit` stuff can be run there. If we have the case where one test needs multiple repos, maybe we should split those up?